doc/nrf/libraries/caf/settings_loader.rst

```diff
@@ -0,0 +1,62 @@
+.. _caf_settings_loader:
+
+CAF: Settings loader module
+###########################
+
+.. contents::
+   :local:
+   :depth: 2
+
+Zephyr's Bluetooth® stack does not load the :ref:`zephyr:settings_api` data on its own.
+Zephyr assumes that the application will call :c:func:`settings_load` after completing all necessary initialization.
+The |settings_loader| of the :ref:`lib_caf` (CAF) is responsible for loading the data from non-volatile memory after all necessary initialization is done.
+
+Configuration
+*************
+
+The following Kconfig options are required:
+
+* :kconfig:`CONFIG_CAF_SETTINGS_LOADER` - This option enables the |settings_loader|.
+* :kconfig:`CONFIG_SETTINGS` - This option enables Zephyr's :ref:`zephyr:settings_api`.
+
+The following Kconfig options are also available for the module:
+
+* :kconfig:`CONFIG_CAF_SETTINGS_LOADER_DEF_PATH`
+* :kconfig:`CONFIG_CAF_SETTINGS_LOADER_USE_THREAD`
+* :kconfig:`CONFIG_CAF_SETTINGS_LOADER_THREAD_STACK_SIZE`
+
+To use the module, you must complete the following requirements:
+
+1. Enable the :kconfig:`CONFIG_CAF_SETTINGS_LOADER` and :kconfig:`CONFIG_SETTINGS` Kconfig options.
+#. Add the configuration file that implements a function :c:func:`get_req_modules` that sets bits of modules that are required to be initialized before settings are loaded.
+
+   For example, the file content could look like follows:
+
+   .. code-block:: c
+
+      #include <caf/events/module_state_event.h>
+
+      static inline void get_req_modules(struct module_flags *mf)
+      {
+              module_flags_set_bit(mf, MODULE_IDX(main));
+      #if CONFIG_CAF_BLE_ADV
+              module_flags_set_bit(mf, MODULE_IDX(ble_adv));
+      #endif
+      };
+
+
+This function is called on the settings loader module initialization.
+After each of modules that sets bit in :c:func:`get_req_modules` are initialized, the |settings_loader| calls :c:func:`settings_load` function and starts loading all the settings from non-volatile memory.
+
+Implementation details
+**********************
+
+Getting required modules is wrapped into a :c:func:`get_req_modules` function due to implementation limitaions.
+
+Settings are by default loaded in the system workqueue context.
```

Comment:
`In Event Manager handler that is invoked in a system workqueue context`.
Keep in mind that Event Manager handlers may be invoked in a dedicated thread at some point.

Reply:
Okay, i will update this.